File: gData.lua
PLUGIN.name = "Player Count"
PLUGIN.author = "Usuariozombie"
PLUGIN.description = "Counts the players connected to the server in a JSON file and displays it on the website."

timer.Create("PlayerCount", 10, 0, function()

    local playerData = {}
    local charData = {}

    for k, v in pairs(player.GetAll()) do
        if v:GetCharacter() then
            table.insert(charData, {
                charname = v:GetCharacter():GetName(),
                faction = ix.faction.Get(v:Team()).name,
                id = v:SteamID64(),
                flags = v:GetCharacter():GetFlags(),
                money = v:GetCharacter():GetMoney(),
                eweapons = v:GetWeapons(),
                model = v:GetModel(),
                health = v:Health(),
                armor = v:Armor(),
                steamid = v:SteamID(),

                -- Get the player's position
                position = {
                    x = v:GetPos().x,
                    y = v:GetPos().y,
                    z = v:GetPos().z
                },
            })
        end
    end

    local playerCount = player.GetCount()


    -- Only send post if player data is not nil
    if playerData then
        if playerCount > 0 then
            HTTP({
                method = "POST",
                url = "https://yourapi.com",
                type = "application/json",
                body = util.TableToJSON({playerCount = playerCount, charData = charData}),
                failed = function(code)
                    print("Failed to send player count to the API. Error code: " .. code)
                end,
                success = function(code, body, headers)
                    print("Player count sent to the API.")
                    print("Response: " .. body)
                end
            })
        else
            HTTP({
                method = "POST",
                url = "https://yourapi.com",
                type = "application/json",
                body = util.TableToJSON({playerCount = playerCount}),
                failed = function(code)
                    print("Failed to send player count to the API. Error code: " .. code)
                end,
                success = function(code, body, headers)
                    print("Player count sent to the API.")
                    print("Response: " .. body)
                end
            })
        end
    else
        print("Player data is nil.")
    end
end)

File: gData.py
# -*- coding: utf-8 -*-

import json, math, requests, time

class JSON():
	def Read(FilePath):
		with open(FilePath, "r", encoding = "utf-8") as JSONFile: Data = json.load(JSONFile)
		JSONFile.close()
		return Data
	def Write(FilePath, Data):
		with open(FilePath, "w") as JSONFile: json.dump(Data, JSONFile, indent = "\t")

APIData = requests.get("https://yourapi.com").json()
CharData = APIData["charData"]
PlayerCount = APIData["playerCount"]

LocalDB = JSON.Read("playerData.json")

for Character in CharData:
	if Character["id"] in LocalDB:
		if Character["charname"] in LocalDB[Character["id"]]:
			if LocalDB[Character["id"]][Character["charname"]]["faction"] != Character["faction"]:
				LocalDB[Character["id"]][Character["charname"]]["faction"] = Character["faction"]

			if LocalDB[Character["id"]][Character["charname"]]["flags"] != Character["flags"]:
				LocalDB[Character["id"]][Character["charname"]]["flags"] = Character["flags"]

			if LocalDB[Character["id"]][Character["charname"]]["health"] != Character["health"]:
				LocalDB[Character["id"]][Character["charname"]]["health"] = Character["health"]

			if LocalDB[Character["id"]][Character["charname"]]["model"] != Character["model"]:
				LocalDB[Character["id"]][Character["charname"]]["model"] = Character["model"]

		else: LocalDB[Character["id"]][Character["charname"]] = Character
			
	else:
		LocalDB[Character["id"]] = {}
		LocalDB[Character["id"]][Character["charname"]] = Character

JSON.Write("playerData.json", LocalDB)